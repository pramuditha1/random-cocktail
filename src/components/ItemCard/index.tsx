import * as React from "react";
import Card from "@mui/material/Card";
import CardActions from "@mui/material/CardActions";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import { ButtonProps } from "@mui/material/Button";
import Lable from "../Lable";
import ButtonComp from "../Button";
import { localization } from "../../locale";

export interface ItemCardPropTypes {
  name: string;
  description: string;
  image: string;
  buttonAction: ButtonProps["onClick"];
}

const ItemCard: React.FC<ItemCardPropTypes> = ({
  name,
  description,
  image,
  buttonAction,
}) => {
  return (
    <Card sx={{ maxWidth: 250, margin: 2 }}>
      <CardMedia
        component="img"
        alt={`${name}-image`}
        height="140"
        image={image}
      />
      <CardContent>
        <Lable variant="h5" text={name} />
        <Lable variant="body2" color="text.secondary" text={description} />
      </CardContent>
      <CardActions>
        <ButtonComp
          variant="outlined"
          text={localization.addToFavourites}
          onClickHandler={buttonAction}
        />
      </CardActions>
    </Card>
  );
};

export default ItemCard;
