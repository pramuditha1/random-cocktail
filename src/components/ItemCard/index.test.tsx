import React from "react";
import { render } from "@testing-library/react";
import ItemCard, { ItemCardPropTypes } from "./index";

test("Header matches snapshot", () => {
  const mockProps: ItemCardPropTypes = {
    name: "name",
    description: "description",
    image:
      "https://www.thecocktaildb.com/images/media/drink/qxuppv1468875308.jpg",
    addToCartHandler: jest.fn(),
  };

  const { asFragment } = render(<ItemCard {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
