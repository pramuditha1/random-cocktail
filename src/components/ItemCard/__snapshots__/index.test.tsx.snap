// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1dcmvbb-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="name-image"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="https://www.thecocktaildb.com/images/media/drink/qxuppv1468875308.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1rwkfxe-MuiTypography-root"
      >
        name
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-186q1z4-MuiTypography-root"
      >
        description
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add to favourites
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
